### PR TITLE
[tls] Allow refreshing TLS configuration information at runtime

### PR DESCRIPTION
**Pull request by smklein**
Adds a `refresh_tls` method to `HttpServer`, which allows TLS information to be updated for a running server.

Fixes https://github.com/oxidecomputer/dropshot/issues/491

### CODE REVIEW

**Comment by jclulow** [comment]
How does the mutex on the acceptor work if there are no inbound connections being made?

**Comment by jclulow** [comment]
Yes that makes sense thanks.  I had missed that we are not blocking at that point waiting on a connection, because of the select loop it's inside that actually deals directly with the system-level sockets.

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -15,6 +15,7 @@
 
 https://github.com/oxidecomputer/dropshot/compare/v0.8.0\...HEAD[Full list of commits]
 
+* https://github.com/oxidecomputer/dropshot/pull/502[#502] Dropshot exposes a `refresh_tls` method to update the TLS certificates being used by a running server.
```
Comment (ahl) [inline CHANGELOG.adoc:18]: I may be mistaken, but I think this is the first breaking change in this list. If that's the case, can you please create two sections as below?

**Review by ahl — COMMENTED** [review]
ship it

File: CHANGELOG.adoc (hunk 1 above)
Comment (smklein) [reply CHANGELOG.adoc]: Sure, updated in https://github.com/oxidecomputer/dropshot/pull/502/commits/219ff7af4890eed9227d1cb9d5cb9931a5ee16d0

**Comment by smklein** [comment]
FYI @davepacheco , since this is a breaking change

**Pull request merged**